### PR TITLE
Added filter when getting donor's purchase value.

### PR DESCRIPTION
**Pull request by emgk**
## Description
Added filter when getting the donor's purchase value.

Status: In-progress.

## How Has This Been Tested?
- UnitTest
- Manually by adding the filter.

## Types of changes
- Introduce new filter when getting purchase value.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/class-give-donor.php [hunk 1]
```diff
@@ -790,6 +790,26 @@ public function get_notes_count() {
 	}
 
 	/**
+	 * Get the purchase value.
+	 *
+	 * @since 1.8.17
+	 *
+	 * @return string|float
+	 */
+	public function get_purchase_value() {
```
Comment (ravinderk) [inline includes/class-give-donor.php]: @emgk Please rename this function to `get_total_donation_amount` and filter name to `give_get_total_donation_amount`

Comment (DevinWalker) [reply includes/class-give-donor.php]: Agreed. The name should be updated to how @ravinderk has mentioned.

Comment (emgk) [reply includes/class-give-donor.php]: @DevinWalker I will update as @ravinderk mentioned, currently working on it.

File: includes/class-give-donor.php [hunk 2]
```diff
@@ -790,6 +790,29 @@ public function get_notes_count() {
...
 	/**
+	 * Get the total donation amount.
+	 *
+	 * @since 1.8.17
+	 *
+	 * @param array $args Pass any additional data.
+	 *
+	 * @return string|float
+	 */
+	public function get_total_donation_amount( $args = array() ) {
```
Comment (ravinderk) [inline includes/class-give-donor.php]: @DevinWalker Please review new function with filter to get total donation amount

**Pull request merged**